lib/bundler/runtime.rb

```diff
@@ -20,29 +20,20 @@ def setup(*groups)
       specs = groups.any? ? @definition.specs_for(groups) : requested_specs
 
       SharedHelpers.set_bundle_environment
-      Bundler.rubygems.replace_entrypoints(specs)
 
-      # Activate the specs
-      load_paths = specs.map do |spec|
+      specs.each do |spec|
         unless spec.loaded_from
           raise GemNotFound, "#{spec.full_name} is missing. Run `bundle install` to get it."
         end
 
         check_for_activated_spec!(spec)
```

Comment:
This method seems very dependent on rubygems, why not moving everything to the new `setup` method?